### ISSUE TITLE
Please support recent versions of QuickCheck and aeson

### ISSUE THREAD

**Issue by peti** [jprupp/haskoin-core#295]
openSUSE distributes your package, but recently it stopped compiling becase we've updated to the Stackage LTS 8.x package set, which causes the following issue during "configure":

~~~
Configuring haskoin-core-0.4.0...
Setup: Encountered missing dependencies:
QuickCheck >=2.6 && <2.9, aeson >=0.7 && <0.12
~~~

**Comment by runeksvendsen** [comment]
Could you try building `master`, and see if that fixes it? As far as I can see, the update I did to LTS 7.18 should make it work for 8.x as well.

**Comment by peti** [comment]
I edited the `resolver` field in `stack.yaml` to `8.2` and ran `stack build haskoin-core` with the following outcome:

~~~
Error: While constructing the build plan, the following exceptions were encountered:

In the dependencies for murmur3-1.0.2:
    cereal-0.5.4.0 must match >=0.5.1 && <0.5.3 (latest applicable is 0.5.2.0)
needed due to haskoin-core-0.4.0 -> murmur3-1.0.2
~~~

**Comment by peti** [comment]
OK, these changes made the build succeed:

~~~
diff --git a/stack.yaml b/stack.yaml
index 749f9eb..3eb7cf3 100644
--- a/stack.yaml
+++ b/stack.yaml
@@ -5,7 +5,7 @@ packages:
 - haskoin-wallet
 extra-deps:
 - daemons-0.2.1
-- murmur3-1.0.2
+- murmur3-1.0.3
 - pbkdf-1.1.1.1
 - secp256k1-0.4.6
-resolver: lts-6.15
+resolver: lts-8.2
~~~

**Comment by runeksvendsen** [comment]
Great!

When I need to build for lts-7.18, I also need to change `murmur3` to 1.0.3, and add some other package, that is not needed for 6.13. It would be nice if Stack had some way to specify these things as a config option, rather than having to edit `stack.yaml` manually for the purpose. But as far as I know this is not the case.

So, for anyone ending up here from a Google search: 

* To build with `lts-8.x`: `murmur3-1.0.2 -> murmur3-1.0.3`
* To build with `lts-7.18`: `murmur3-1.0.2 -> murmur3-1.0.3` + add `esqueleto-2.5.1`

**Comment by peti** [comment]
Will you release a new version that remedies the issue with QuickCheck and aeson versions?

**Comment by runeksvendsen** [comment]
@plaprade @xenog Could one of you guys take a look at this? I don't have Hackage access for haskoin. Perhaps it's time for 0.4.1?

**Comment by jprupp** [comment]
I will take a look over the weekend. I made you release manager in hackage for _haskoin-core_, _haskoin-node_ and _haskoin-wallet_.

**Comment by wyager** [comment]
Ditto here, I'm getting the following with lts-8.5:

```
--  Failure when adding dependencies:
      QuickCheck: needed (>=2.6 && <2.9), 2.9.2 found (latest applicable is 2.8.2)
      aeson: needed (>=0.7 && <0.12), 1.0.2.1 found (latest applicable is 0.11.3.0)
      murmur3: needed (>=1.0 && <1.1), stack configuration has no specified version (latest applicable is 1.0.3)
      pbkdf: needed (>=1.1 && <1.2), stack configuration has no specified version (latest applicable is 1.1.1.1)
      secp256k1: needed (>=0.4 && <0.5), stack configuration has no specified version (latest applicable is 0.4.6)
    needed for package haskoin-core-0.4.0
```

`stack solver` seems to find a functional build plan on its second try, but it'd be better to get Haskoin-core back in the LTS solvers.

**Comment by runeksvendsen** [comment]
@xenog After pull request #297 is merged, the master branch should be ready for a 0.4.1 release. I will wait a week -- unless I hear from you before then, telling me to either go ahead or wait a bit --  until merging #297, after which I will release this as haskoin-{core,node,wallet} 0.4.1. 

Also, I would like to add both haskoin-node and haskoin-wallet to Stackage, in addition to haskoin-core. Which email address should I use when re-adding these packages to the [Stackage build-constraints.yaml](https://github.com/fpco/stackage/blob/master/MAINTAINERS.md#adding-a-package)? I plan on using @haskoin/maintainers as the GitHub user name, but I'm not sure which email address to use.

**Comment by jprupp** [comment]
Changes have been uploaded and a new version released as 0.4.2. Use your email address Rune as Stackage maintainer. Alternatively use my email xenog@protonmail.com.
